notes.go

```diff
@@ -71,9 +72,66 @@ func (srv *groupsAPI) UpdateGroupNote(ctx context.Context, in *accountsv1.Update
 }
 
 func (srv *groupsAPI) GetGroupNote(ctx context.Context, in *accountsv1.GetGroupNoteRequest) (*accountsv1.GetGroupNoteResponse, error) {
-	return nil, errors.New("not implemented")
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = srv.memberRepo.Get(ctx, &models.MemberFilter{AccountID: &accountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "could not validate get groupNotes request")
+	}
+
+	err = validators.ValidateGetGroupNote(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	note, err := srv.noteRepo.Get(ctx, &models.GroupNoteFilter{NoteID: in.NoteId, GroupID: in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	groupNote := accountsv1.GroupNote{AuthorAccountId: note.AuthorID, NoteId: note.NoteID, Title: note.Title}
+	return &accountsv1.GetGroupNoteResponse{Note: &groupNote}, nil
 }
 
 func (srv *groupsAPI) ListGroupNotes(ctx context.Context, in *accountsv1.ListGroupNotesRequest) (*accountsv1.ListGroupNotesResponse, error) {
-	return nil, errors.New("not implemented")
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	err = validators.ValidateListGroupNote(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "could not validate list groupNotes request")
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = srv.memberRepo.Get(ctx, &models.MemberFilter{AccountID: &accountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to get member from group_id")
```

Comment:
Should be `statusFromModelError()`. Super important for the right HTTP code to be chosen + clarity of error messages.